### DISCUSSION TITLE
Achieving Successful Git Push After Password Prompt for GitHub HTTPS Repository Clone

### DISCUSSION THREAD

**Discussion by atanugiri** [community/community#69055] [New to GitHub] [answered]
After cloning a GitHub repository to my local computer via HTTPS, I made code modifications locally, performed a 'git add' and 'git commit.' However, when attempting to execute 'git push origin main,' I encountered a prompt for my username and password. I initially tried entering the credentials saved in Safari, but I received an error. How can I resolve this issue? I also attempted to paste the password from both '/.ssh/id_rsa' and '/.ssh/id_rsa.pub' without success.

**Comment by rnldvrgl** [comment, marked as answer]
Generate a Personal Access Token:

    Go to your GitHub account settings by clicking on your profile picture in the upper right corner and selecting "Settings."

    In the left sidebar, click on "Developer settings."

    Click on "Personal access tokens."

    Click the "Generate token" button.

    Provide a token description (e.g., "Git operations").

    Choose the scopes/permissions you need. For pushing code to repositories, the "repo" scope is necessary.

    Click "Generate token."

    Copy the generated token to your clipboard. Be careful with it, as you won't be able to see it again. If you lose it, you'll need to regenerate a new token.


Update Git Credentials:

    Open your terminal.

    Use the following command to update your Git credentials with the personal access token you generated:
    
    git config --global credential.helper store



Push Your Changes:

    Now, when you run git push origin main, Git will prompt you for your GitHub username, but you should enter your GitHub username as usual and, for the password, paste the personal access token you generated in step 1.

    After entering your credentials, Git should successfully push your changes to the remote repository.

**Comment by atanugiri** [comment]
Thank you! That was very helpful.

**Reply by rnldvrgl** [reply]
No worries, hope you can now successfully push your changes. If this answer your question, you can mark this as answered so it will help others that has the same issue as yours, have a great day!
